### PR TITLE
Feature/big query fix

### PR DESCRIPTION
**Pull request by siddhant3030**
## Summary
 Target issue is #2247 
 This is to solve the issue with biguery credentials. It would not accept any random character.

## Checklist
  Before submitting a pull request, please ensure that you mark these task.

- [ ] Ran `mix setup` in the repository root and test.
- [ ] If you've fixed a bug or added code that is tested and has test cases. 
- [ ] In the case of adding a new API, you added a **postman** request for that.
- [ ] Coding standard and conventions are followed. https://docs.google.com/document/d/1rfU33IjS-ioiIH0TBWpxoLsdyyYdI1tDrqr5HCRIb98/edit?usp=sharing

## Notes
 Please add here if any other information is required for the reviewer.

### CODE REVIEW

File: lib/glific/partners.ex [hunk 1]
```diff
@@ -844,11 +844,40 @@ defmodule Glific.Partners do
...
+        {:ok, credential}
+
+      {:error, _error} ->
+        {:error, "Invalid Credentials"}
+    end
+  end
+
+  def credential_update_callback(organization, _credential, "google_cloud_storage") do
+    case GCS.refresh_gcs_setup(organization.id) do
+      {:ok, a} -> {:ok, a}
```
Comment (AkhileshNegi) [inline lib/glific/partners.ex]: what's `a`?
can we return credential?

File: lib/glific/partners.ex [hunk 2]
```diff
@@ -844,11 +844,40 @@ defmodule Glific.Partners do
...
+      {:error, _error} -> {:error, "Invalid Credentials"}
+    end
+  end
+
+  def credential_update_callback(organization, _credential, "dialogflow") do
+    case Glific.Dialogflow.get_intent_list(organization.id) do
+      {:ok, a} -> {:ok, a}
```
Comment (AkhileshNegi) [inline lib/glific/partners.ex]: what's `a`?
can we return credential?

File: lib/glific/partners.ex [hunk 3]
```diff
@@ -816,7 +816,7 @@ defmodule Glific.Partners do
   """
 
   @spec update_credential(Credential.t(), map()) ::
-          {:ok, Credential.t()} | {:error, Ecto.Changeset.t()}
+          {:ok, any} | {:error, any}
```
Comment (AkhileshNegi) [inline lib/glific/partners.ex]: I think it still return {:ok, Credential.t()}

File: lib/glific/third_party/bigquery/bigquery.ex [hunk 4]
```diff
@@ -72,26 +72,41 @@ defmodule Glific.BigQuery do
...
-    :ok
   end
 
   @doc false
-  @spec fetch_bigquery_credentials(non_neg_integer) :: nil | tuple
+  @spec fetch_bigquery_credentials(non_neg_integer) :: nil | {:ok, any} | {:error, any}
   def fetch_bigquery_credentials(organization_id) do
     organization = Partners.organization(organization_id)
     org_contact = organization.contact
+    info = %{org_contact: org_contact, organization_id: organization_id}
```
Comment (AkhileshNegi) [inline lib/glific/third_party/bigquery/bigquery.ex]: why do we need to put `org_contact` and `organization_id` in map when in line 104 we only fetching the exact values?
can we use `decode_bigquery_credential(credentials, org_contact, organization_id)` instead

File: lib/glific/third_party/gcs/gcs.ex [hunk 5]
```diff
@@ -32,30 +32,39 @@ defmodule Glific.GCS do
...
-    organization_id
-    |> insert_gcs_jobs()
+    case insert_gcs_jobs(organization_id) do
+      {:ok, gcs} -> {:ok, gcs}
+      {:error, error} -> {:error, error}
+    end
 
-    :ok
+    # organization_id
+    # |> insert_gcs_jobs()
```
Comment (AkhileshNegi) [inline lib/glific/third_party/gcs/gcs.ex]: remove the commented lines

File: lib/glific/third_party/gcs/gcs.ex [hunk 6]
```diff
@@ -32,30 +32,39 @@ defmodule Glific.GCS do
...
+          {:ok, organization} ->
+            {:ok, organization}
 
-    :ok
+          {:error, error} ->
+            {:error, error}
+        end
+
+        # %GcsJob{organization_id: organization_id}
+        # |> Repo.insert!()
```
Comment (AkhileshNegi) [inline lib/glific/third_party/gcs/gcs.ex]: remove commented lines

File: lib/glific/third_party/gcs/gcs.ex [hunk 7]
```diff
@@ -32,30 +32,39 @@ defmodule Glific.GCS do
...
-        |> Repo.insert!()
-    end
+        case %GcsJob{} |> GcsJob.changeset(organization_id) |> Repo.insert() do
+          {:ok, organization} ->
+            {:ok, organization}
 
-    :ok
+          {:error, error} ->
+            {:error, error}
+        end
```
Comment (AkhileshNegi) [inline lib/glific/third_party/gcs/gcs.ex]: can you revert these changes and create a separate PR stating what problem it is solving

Comment (siddhant3030) [reply lib/glific/third_party/gcs/gcs.ex]: It is being used by partner.ex file. Look at the function there credential_update_callback. It expect tuple instead of just :ok

File: lib/glific/third_party/gcs/gcs.ex (hunk 6 above)
Comment (siddhant3030) [reply lib/glific/third_party/gcs/gcs.ex]: done

File: lib/glific/third_party/gcs/gcs.ex (hunk 5 above)
Comment (siddhant3030) [reply lib/glific/third_party/gcs/gcs.ex]: done

File: lib/glific/third_party/bigquery/bigquery.ex (hunk 4 above)
Comment (siddhant3030) [reply lib/glific/third_party/bigquery/bigquery.ex]: This also we can do. But I was avoiding 3 arguments in the function. I'll change this

File: lib/glific/partners.ex (hunk 2 above)
Comment (siddhant3030) [reply lib/glific/partners.ex]: done

File: lib/glific/partners.ex (hunk 3 above)
Comment (siddhant3030) [reply lib/glific/partners.ex]: done

File: lib/glific/third_party/gcs/gcs.ex [hunk 8]
```diff
@@ -32,30 +32,33 @@ defmodule Glific.GCS do
...
   def refresh_gcs_setup(organization_id) do
     Logger.info("refresh GCS setup for org_id: #{organization_id}")
 
-    organization_id
-    |> insert_gcs_jobs()
-
-    :ok
+    case insert_gcs_jobs(organization_id) do
+      {:ok, gcs} -> {:ok, gcs}
+      {:error, error} -> {:error, error}
```
Comment (AkhileshNegi) [inline lib/glific/third_party/gcs/gcs.ex]: if we are anyway returning the same tuple can we just use
`insert_gcs_jobs(organization_id)`
instead of case statement from line 39-41

File: lib/glific/third_party/gcs/gcs.ex (hunk 7 above)
Comment (AkhileshNegi) [reply lib/glific/third_party/gcs/gcs.ex]: Here also as we are returning the same tuple so adding case statement does not make sense
